tests/components/logger/test_init.py

```diff
@@ -1,143 +1,190 @@
 
-    await hass.services.async_call(
-        logger.DOMAIN, "set_default_level", {"level": "warning"}
+async def test_can_set_level(hass):
+    """Test logger propagation."""
+
+    assert await async_setup_component(
+        hass,
+        "logger",
+        {
+            "logger": {
+                "logs": {
+                    CONFIGED_NS: "warning",
+                    f"{CONFIGED_NS}.info": "info",
+                    f"{CONFIGED_NS}.debug": "debug",
+                    HASS_NS: "warning",
+                    COMPONENTS_NS: "info",
+                    ZONE_NS: "debug",
+                    GROUP_NS: "info",
+                },
+            }
+        },
     )
-    await hass.async_block_till_done()
-
-    assert not log_filter.filter(RECORD("test", logging.DEBUG))
 
-    assert logging.getLogger("").isEnabledFor(logging.DEBUG) is False
-    assert logging.getLogger("").isEnabledFor(logging.WARNING) is True
+    assert logging.getLogger(UNCONFIG_NS).level == logging.NOTSET
+    assert logging.getLogger(UNCONFIG_NS).isEnabledFor(logging.CRITICAL) is True
+    assert (
+        logging.getLogger(f"{UNCONFIG_NS}.any").isEnabledFor(logging.CRITICAL) is True
+    )
+    assert (
+        logging.getLogger(f"{UNCONFIG_NS}.any.any").isEnabledFor(logging.CRITICAL)
+        is True
+    )
 
-    await hass.services.async_call(logger.DOMAIN, "set_level", {"test": "debug"})
-    await hass.async_block_till_done()
+    assert logging.getLogger(CONFIGED_NS).isEnabledFor(logging.DEBUG) is False
+    assert logging.getLogger(CONFIGED_NS).isEnabledFor(logging.WARNING) is True
+    assert logging.getLogger(f"{CONFIGED_NS}.any").isEnabledFor(logging.WARNING) is True
+    assert (
+        logging.getLogger(f"{CONFIGED_NS}.any.any").isEnabledFor(logging.WARNING)
+        is True
+    )
+    assert logging.getLogger(f"{CONFIGED_NS}.info").isEnabledFor(logging.DEBUG) is False
+    assert logging.getLogger(f"{CONFIGED_NS}.info").isEnabledFor(logging.INFO) is True
+    assert (
+        logging.getLogger(f"{CONFIGED_NS}.info.any").isEnabledFor(logging.DEBUG)
+        is False
+    )
+    assert (
+        logging.getLogger(f"{CONFIGED_NS}.info.any").isEnabledFor(logging.INFO) is True
+    )
+    assert logging.getLogger(f"{CONFIGED_NS}.debug").isEnabledFor(logging.DEBUG) is True
+    assert (
+        logging.getLogger(f"{CONFIGED_NS}.debug.any").isEnabledFor(logging.DEBUG)
+        is True
+    )
 
-    assert log_filter.filter(RECORD("test", logging.DEBUG))
+    assert logging.getLogger(HASS_NS).isEnabledFor(logging.DEBUG) is False
+    assert logging.getLogger(HASS_NS).isEnabledFor(logging.WARNING) is True
 
-    assert logging.getLogger("").isEnabledFor(logging.DEBUG) is True
+    assert logging.getLogger(COMPONENTS_NS).isEnabledFor(logging.DEBUG) is False
+    assert logging.getLogger(COMPONENTS_NS).isEnabledFor(logging.WARNING) is True
+    assert logging.getLogger(COMPONENTS_NS).isEnabledFor(logging.INFO) is True
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5844602)